File: scala/problem-00019.scala
/*
[minhuang@mocha:scala]$ time scala problem-00019.scala 
171

real  0m1.463s
user  0m0.308s
sys 0m0.080s
*/
import java.util.Calendar
import java.util.Calendar._

var cal = Calendar.getInstance
var sundays = 0

for (year <- 1901 to 2000;
     month <- 0 to 11) {
  cal.set(year, month, 1)
  if (cal.get(DAY_OF_WEEK) == SUNDAY) sundays += 1
}

println(sundays)

File: src/javascript/problem-00019.js
// [minhuang@mocha:javascript(master*)]$ time js -f problem-00019.js 
// 171
//
// real 0m0.041s
// user 0m0.024s
// sys  0m0.012s
function gen(a, b, c, d) {
  for (let i = a; i <= b; i++)
    for (let j = c; j <= d; j++)
      yield [i, j];
}

let sundays = 0;
let months = [
  [year, month] for each ([year, month] in gen(1901, 2000, 1, 12))
];
let result = months.filter(function (e) {
    let [year, month] = e;
    let d = new Date(year, month - 1, 1);
    return d.getDay() === 0;
  }).length;
print(result);
